gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -117,8 +117,9 @@ public CopyDataPublisher(State state) throws IOException {
     this.fs = FileSystem.get(URI.create(uri), WriterUtils.getFsConfiguration(state));
 
     FileAwareInputStreamDataWriterBuilder.setJobSpecificOutputPaths(state);
-
-    this.writerOutputDir = new Path(state.getProp(ConfigurationKeys.WRITER_OUTPUT_DIR));
+    // If directories are sharded by dataset, initialize writers using workunit state instead
+    this.writerOutputDir = state.getPropAsBoolean(ConfigurationKeys.USE_DATASET_LOCAL_WORK_DIR) ?
```

Comment:
Shouldn't WRITER_OUTPUT_DIR be set inside the handler? If so, we don't need these changes here, do we?

Reply:
I don't understand the explanation. What I am suggesting is to have the ConfigurationKeys#WRITER_OUTPUT_DIR being overwritten inside the DatasetHandler impl, which would avoid the need for this change.